### PR TITLE
procMount: fix default value documentation

### PR DESCRIPTION
**Pull request by aborrero**
The 'DefaultProcMount' string is the name of the internal variable, but the actual value is 'Default'.

#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:

Documentation update. Previous to this PR, it was inaccurate. This PR fixes it.

#### Which issue(s) this PR fixes:

None.

#### Special notes for your reviewer:

None.

#### Does this PR introduce a user-facing change?

Maybe? Documentation update.

```release-note
Fix the documentation for the default value of the procMount entry in the pod securityContext.
The documentation was previously using the name of the internal variable 'DefaultProcMount' rather than the actual value 'Default'.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

None.

### CODE REVIEW

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -6203,7 +6203,7 @@ type SecurityContext struct {
 	// +optional
 	AllowPrivilegeEscalation *bool
 	// ProcMount denotes the type of proc mount to use for the containers.
-	// The default is DefaultProcMount which uses the container runtime defaults for
+	// The default value is Default which uses the container runtime defaults for
```
Comment (aojea) [inline pkg/apis/core/types.go:6206]: good catch, however, these are the internal types, the ones we need to update for the documentation are here

https://github.com/kubernetes/kubernetes/blob/eb66365bc4a84eca9b7f4ca6436535bc78d46648/staging/src/k8s.io/api/core/v1/types.go#L7454-L7460

so once you update both places, do `make update` locally to regenerate the documentation and push all together

Comment (aborrero) [reply pkg/apis/core/types.go:6206]: thanks for the pointers, I did as you said.

**Pull request merged**